src/utilities/separateOperations.js

```diff
@@ -24,18 +25,22 @@ export function separateOperations(
   documentAST: DocumentNode
 ): { [operationName: string]: DocumentNode } {
 
-  const operations = [];
+  const definitions: DefinitionMap = Object.create(null);
   const depGraph: DepGraph = Object.create(null);
   let fromName;
+  let idx = 0;
 
-  // Populate the list of operations and build a dependency graph.
+  // Populate the list of definitions and build a dependency graph.
   visit(documentAST, {
     OperationDefinition(node) {
-      operations.push(node);
       fromName = opName(node);
+      definitions[fromName] = { idx, node };
+      ++idx;
     },
     FragmentDefinition(node) {
       fromName = node.name.value;
+      definitions[fromName] = { idx, node };
```

Comment:
There's currently nothing to guarantee that an operation and fragment definition will always have distinct names. It would be safer to have separate `operationDefinitions` and `fragmentDefinitions` maps